ank/src/cli_commands/set_state.rs

```diff
@@ -16,94 +16,133 @@ use common::{
-                runtime: "".to_string(),
-                runtime_config: "".to_string(),
-                ..Default::default()
-            };
+            complete_state.desired_state.workloads.insert(
+                path.parts()[WORKLOAD_NAME_POSITION].to_string(),
+                StoredWorkloadSpec::default(),
+            );
+        }
+    }
 
-            complete_state
-                .desired_state
-                .workloads
-                .insert(path.parts()[2].to_string(), stored_workload);
+    complete_state
+}
+
+// [impl->swdd~cli-supports-yaml-to-set-desired-state~1]
+async fn process_inputs<R: Read>(reader: R, state_object_file: &str) -> Result<Object, CliError> {
+    match state_object_file {
+        "-" => {
+            let stdin = io::read_to_string(reader).map_err(|error| {
+                CliError::ExecutionError(format!(
+                    "Could not read the state object from stdin.\nError: '{}'",
+                    error
+                ))
+            })?;
+            let value: serde_yaml::Value = serde_yaml::from_str(&stdin).map_err(|error| {
+                CliError::YamlSerialization(format!(
+                    "Could not convert stdin input to yaml.\nError: '{}'",
+                    error
+                ))
+            })?;
+            Ok(Object::try_from(&value)?)
+        }
+        _ => {
+            let state_object_data =
+                read_file_to_string(state_object_file.to_string()).map_err(|error| {
+                    CliError::ExecutionError(format!(
+                        "Could not read the state object file.\nError: '{}'",
+                        error
+                    ))
+                })?;
+            let value: serde_yaml::Value =
+                serde_yaml::from_str(&state_object_data).map_err(|error| {
+                    CliError::YamlSerialization(format!(
+                        "Could not convert state object file to yaml Value.\nError: '{}'",
+                        error
+                    ))
+                })?;
+            Ok(Object::try_from(&value)?)
         }
     }
 }
 
+// [impl->swdd~cli-supports-yaml-to-set-desired-state~1]
+fn overwrite_using_field_mask(
+    mut complete_state_object: Object,
+    object_field_mask: &Vec<String>,
+    temp_obj: &Object,
+) -> Result<Object, CliError> {
+    for field_mask in object_field_mask {
+        let path: Path = field_mask.into();
+
+        complete_state_object
+            .set(
+                &path,
+                temp_obj
+                    .get(&path)
+                    .ok_or(CliError::ExecutionError(format!(
+                        "Specified update mask '{field_mask}' not found in the input config.",
+                    )))?
+                    .clone(),
+            )
+            .map_err(|err| CliError::ExecutionError(err.to_string()))?;
+    }
+
+    Ok(complete_state_object)
+}
+
 impl CliCommands {
     pub async fn set_state(
```

Comment:
I think we need a linkage to this function:
```suggestion
    // [impl->swdd~cli-supports-yaml-to-set-desired-state~1]
    pub async fn set_state(
```